bookkeeper-server/src/main/java/org/apache/bookkeeper/meta/AbstractHierarchicalLedgerManager.java

```diff
@@ -63,7 +63,11 @@ void asyncProcessLevelNodes(
         zk.sync(path, new AsyncCallback.VoidCallback() {
             @Override
             public void processResult(int rc, String path, Object ctx) {
-                if (rc != Code.OK.intValue()) {
+                if (rc == Code.NONODE.intValue()) {
```

Comment:
These commits here seem not related to "decouple metaformat cmd". I would suggest they should go to a separate issue.

Reply:
ok..would move the 3rd commit to a new PR. But Bug1 in Issue #1014, would be fixed in this PR, since it is getting fixed as part of my changes for this PR. But Bug2 in Issue #1014 and testcases for both the bugs will be in new PR (which is currently 3rd commit).